core/templates/pages/exploration-player-page/learner-experience/conversation-skin.component.ts

```diff
@@ -1332,141 +1332,160 @@ export class ConversationSkinComponent {
           });
 
           if (remainOnCurrentCard) {
-            // Stay on the same card.
-            this.numberOfIncorrectSubmissions++;
-            this.hintsAndSolutionManagerService.recordWrongAnswer();
-            this.conceptCardManagerService.recordWrongAnswer();
-            this.playerTranscriptService.addNewResponse(feedbackHtml);
-            let helpCardAvailable = false;
-            if (feedbackHtml &&
-                !this.displayedCard.isInteractionInline()) {
-              helpCardAvailable = true;
-            }
-
-            if (helpCardAvailable) {
-              this.playerPositionService.onHelpCardAvailable.emit({
-                helpCardHtml: feedbackHtml,
-                hasContinueButton: false
-              });
-            }
-            if (missingPrerequisiteSkillId) {
-              this.displayedCard.markAsCompleted();
-              this.conceptCardBackendApiService.loadConceptCardsAsync(
-                [missingPrerequisiteSkillId]
-              ).then((conceptCardObject) => {
-                this.conceptCard = conceptCardObject[0];
-                if (helpCardAvailable) {
-                  this.playerPositionService.onHelpCardAvailable.emit({
-                    helpCardHtml: feedbackHtml,
-                    hasContinueButton: true
-                  });
-                }
-              });
-            }
-            if (refreshInteraction) {
-              // Replace the previous interaction with another of the
-              // same type.
-              this._nextFocusLabel = (
-                this.focusManagerService.generateFocusLabel());
-              this.playerTranscriptService.updateLatestInteractionHtml(
-                this.displayedCard.getInteractionHtml() +
-                this._getRandomSuffix());
-            }
-
-            this.redirectToRefresherExplorationConfirmed = false;
-
-            if (refresherExplorationId) {
-              // TODO(bhenning): Add tests to verify the event is
-              // properly recorded.
-              let confirmRedirection = () => {
-                this.redirectToRefresherExplorationConfirmed = true;
-                this._recordLeaveForRefresherExp(refresherExplorationId);
-              };
-              this.explorationSummaryBackendApiService
-                .loadPublicExplorationSummariesAsync([refresherExplorationId])
-                .then((response) => {
-                  if (response.summaries.length > 0) {
-                    this.refresherExplorationConfirmationModalService.
-                      displayRedirectConfirmationModal(
-                        refresherExplorationId, confirmRedirection);
-                  }
-                });
-            }
-            this.focusManagerService.setFocusIfOnDesktop(this._nextFocusLabel);
-            this.scrollToBottom();
+            this.remainOnCurrentCard(
```

Comment:
I think there are two issues with this method name -- (a) it isn't really an action, (b) it duplicates the variable name directly above. 

What does this function actually do? Maybe giveFeedbackAndStayOnCurrentCard() or something like that?

Reply:
A very good name, thanks!